### PR TITLE
Add PositionedCard field

### PR DESCRIPTION
**Pull request by FadhlanR**
Basically, the idea is to have a `FieldDef` where we can set the position for a card. To test this field, I used the `GardenDesign` card by @burieberry. I replaced the `grid` field in the `GardenDesign` card with `PositionedCard`.

https://github.com/cardstack/boxel/assets/12637010/ddd8553c-d2d9-440e-a371-c7d5a5cc3665

### CODE REVIEW

File: packages/drafts-realm/garden-design.gts [hunk 1]
```diff
@@ -218,27 +184,70 @@ class Isolated extends Component<typeof GardenDesign> {
...
+          (el) => this.composeId(el.position.x, el.position.y) === id,
         );
-        return [id, maybeOption ? maybeOption[1] : null];
+        return [id, maybeOption ? (maybeOption.card as GardenItem) : null];
       }),
     );
     this.updateGridModel();
+    let type = {
+      module: 'http://localhost:4201/drafts/garden-design',
+      name: 'GardenItem',
```
Comment (FadhlanR) [inline packages/drafts-realm/garden-design.gts]: I need a query using a type filter, but I don't know how to make the module URL work in all environments. I believe this module URL works only locally. Does anyone know how to make the module URL environment-agnostic?

Comment (tintinthong) [reply packages/drafts-realm/garden-design.gts]: Would something like this help https://github.com/cardstack/boxel/pull/1368/files#diff-32330c02279a76cb3e85d743f9cc98dd41813d5949eef273e7cd50dcc0b838a2R41-R59?

The ai bot return a relative module path for me. So I need to resolve it and you can specify a realm of choice too

File: packages/base/position.gts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import NumberField from './number';
+import { contains, FieldDef, field, CardDef, linksTo } from './card-api';
+
+export class Position extends FieldDef {
```
Comment (jurgenwerk) [inline packages/base/position.gts]: Is the Position field meant solely to express integer grid coordinates (x,y)? If so, maybe we can consider a more specific name for this field. For example if I saw a Position field I would wonder what kind of a position it is.

File: packages/drafts-realm/garden-design.gts [hunk 3]
```diff
@@ -249,26 +258,51 @@ class Isolated extends Component<typeof GardenDesign> {
...
   };
 
+  composeId(x: number, y: number) {
+    return `r${x}-c${y}`;
+  }
+
+  decomposeId(id: string): { x: number; y: number } {
+    let arr = id.replace('r', '').replace('c', '').split('-');
+    if (!arr[0] || !arr[1]) {
+      throw new Error('Unknown id format');
```
Comment (jurgenwerk) [inline packages/drafts-realm/garden-design.gts]: Would be good to be more exact, i.e 
```throw new Error(`Unknown id format: ${id}. Expected format: rx-cy`);``` 

(I wanted to make this as a code suggestion but for some reason i can't find the suggestion button  🤷‍♂️)

**Review by jurgenwerk — APPROVED** [review]

File: packages/drafts-realm/garden-design.gts (hunk 1 above)
Comment (FadhlanR) [reply packages/drafts-realm/garden-design.gts]: Nvm, I can just use `_cardType` for this. [412140e](https://github.com/cardstack/boxel/pull/1370/commits/412140e0bc3c5791f9d9f3259b5aa720c80ca05c)

**Pull request merged**